### PR TITLE
Server Creates Area

### PR DESCRIPTION
**Pull request by Melestyl**
The server is now able to create multiple areas. TO BE DONE: Add the created area into a list of known areas (to send to clients)

### CODE REVIEW

File: Common/common.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * @struct area
+ * @brief Area structure
+ */
+struct area {
+	char name[16]; // name of the area
+	enum type type; // desk or meeting room
+	pthread_mutex_t mutex; // to prevent any conflicts
+	int shared_memory; // user pid or null
+	char path[120]; // path to shared memory (name of the area)
```
Comment (3Dai1) [inline Common/common.h]: name of the area ?

File: Common/common.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#include <stdio.h>
+#include <string.h>
+
+char* read_config(char* path_name, char* searched_key, char* found_value) {
+	char key[100], value[100], line[200];
```
Comment (3Dai1) [inline Common/common.c]: can you update 100 or 200 into some constant

File: Common/common.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	FILE* file = fopen(path_name, "r");
+
+	// Reading config file
+	if (file == NULL) {
+		printf("Failed to open file\n");
+		return NULL;
+	}
+
+	// Reading line by line
+	while (fgets(line, 200, file) != NULL) {
```
Comment (3Dai1) [inline Common/common.c]: can you update 200 into constant

File: Server/server.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#include "server.h"
+
+area_t* create_area(enum type type, char* name) {
+	static unsigned shared_memory_id = 0;
+	char shared_memory_prefix[100], project_id_str[100], shared_memory_path[120];
```
Comment (3Dai1) [inline Server/server.c]: why 100 ,100 or 120 ? 
Can you replace to constant ? And can you justifie these number ?

File: Common/common.h (hunk 1 above)
Comment (Melestyl) [reply Common/common.h]:
```suggestion
	char path[120]; // path to shared memory (unique name of the shared_memory)
```

**Review by 3Dai1 — APPROVED** [review]

**Pull request merged**